src/lib/hooks/use-best-quote/index.ts

```diff
@@ -336,6 +336,15 @@ export const useBestQuote = (
     quoteRedemption,
   ])
 
+  useEffect(() => {
+    // Reset quotes
+    setQuote0x(null)
+    setQuoteFlashmint(null)
+    setQuoteIssuance(null)
+    setQuoteRedemption(null)
+    setQuoteResults(defaultResults)
+  }, [chainId])
```

Comment:
Actual fix. Started the below first when I realized the fix could be more simple. 

Will continue on refactoring quote results (and integrating it) later but pushing to first get this fix out.